source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -53,16 +55,30 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
   }
   cluster_info_ = cluster->info();
 
-  auto& resource = cluster_info_->resourceManager(route_entry->priority()).pendingRequests();
-  if (!resource.canCreate()) {
+  const auto& dns_cache_resource_manager = config_->cache().dnsCacheResourceManager();
```

Comment:
Instead of leaking the resource manager and the stats from the cache interface, you should be just calling a method called something like `cache().canCreateRequest()` which handles all of the functionality internally including stats.

Reply:
Overcame this with `CircuitBreakersHandler`. It will be useful to implement this in sni_dynamic_forward_proxy.